lib/utils.js

```diff
@@ -141,16 +145,34 @@ exports.getXpiInfoFromManifest = getXpiInfoFromManifest;
 function getXpiInfo(xpiPath) {
   return extractXPI(xpiPath)
     .then(function(tmpXPI) {
-      var manifest = {};
-      try {
-        manifest = require(path.join(tmpXPI.path, "package.json"));
-      } catch (e) {}
-
-      tmpXPI.remove();
+      var getInfo;
+      var packageFile = path.join(tmpXPI.path, "package.json");
+      var installRdfFile = path.join(tmpXPI.path, "install.rdf");
 
-      // TODO: when there's no manifest, parse install.rdf to get info.
+      if (fileExists(packageFile)) {
+        getInfo = getXpiInfoFromManifest(require(packageFile));
+      } else if (fileExists(installRdfFile)) {
+        getInfo = nodefn.call(fs.readFile, installRdfFile)
+          .then(function(data) {
+            return getXpiInfoFromInstallRdf(data);
+          });
+      } else {
+        getInfo = when.promise(function() {
```

Comment:
I think you could use `when.reject(new Error("Cannot get info: no manifest found in this XPI"));` here.